### DISCUSSION TITLE
Help me

### DISCUSSION THREAD

**Discussion by mondayapp** [codemagic-ci-cd/codemagic-docs#2664] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Online build

### Expected results

Apple connect OK

Saved export options to /Users/builder/export_options.plist
> flutter build ipa --release --export-options-plist /Users/builder/export_options.plist

### Actual results

Running pod install...                                             44.9s
Running Xcode build...                                          
Xcode archive done.                                          7.6s
Failed to build iOS app
Error (Xcode): Provisioning profile "XXXXXXXXXXX ios_app_development 1xxxxxxxxxxxxx" doesn't support the Push Notifications and Sign in with Apple capability.
/Users/builder/clone/ios/Runner.xcodeproj

Error (Xcode): Provisioning profile "XXXXXXXXXXXr ios_app_development 1xxxxxxxxxxxxx" doesn't include the aps-environment and com.apple.developer.applesignin entitlements. Profile qualification is using entitlement definitions that may be out of date. Connect to network to update.
/Users/builder/clone/ios/Runner.xcodeproj

It appears that there was a problem signing your application prior to installation on the device.

Verify that the Bundle Identifier in your project is your signing id in Xcode
  open ios/Runner.xcworkspace

Also try selecting 'Product > Build' to fix the problem.
Encountered error while archiving for device.

Build failed :|
Failed to build for iOS

### Build id (optional)

66ac52a6eb768058c5d507fe

**Comment by dam-ease** [comment, marked as answer]
Hello, from your logs, I see these:

```
Error (Xcode): Provisioning profile "Monday User ios_app_development 1722568845" doesn't support the Push Notifications and Sign in with Apple capability.
/Users/builder/clone/ios/Runner.xcodeproj

Error (Xcode): Provisioning profile "Monday User ios_app_development 1722568845" doesn't include the aps-environment and com.apple.developer.applesignin entitlements. Profile qualification is using entitlement definitions that may be out of date. Connect to network to update.
/Users/builder/clone/ios/Runner.xcodeproj
```


Please enable the mentioned capability in the Apple Developer account > identifiers, re-save your provisioning profile, upload to Codemagic and run the build again.

**Reply by AnaNedicGithub** [reply]
Hi @dam-ease, thank you - that solved the current issue.

Unfortunately I still can't upload .ipa because publishing fails.

Could you please check the logs and tell me why? New build number is: **66b222faed118da4ae18bf53**

**Reply by dam-ease** [reply]
Hello @AnaNedicGithub,
The error you're encountering indicates that the permission level assigned to the App Store Connect API key is insufficient. It needs to have either the "App Manager" or "Admin" permission level. Can you please check this?

**Reply by AnaNedicGithub** [reply]
Hi @dam-ease , thank you for your help! My ios build works now :)
(The issue was indeed in the App store Connect API Key, specifically wrong issuer id. Permission level App Manager was sufficient.)
